### PR TITLE
Added prefixed flex class & deprecated scss file

### PR DESCRIPTION
**Pull request by emilybrick**
Added deprecated.scss file: for classes we feel should be deprecated before cold-removing them
Moved the deprecated page messaging styles to that file ☝️ 
Added the prefixed flex display class
Moved non-prefixed flex display class to the new fancy deprecated.scss file (kind of an oxymoron)

Added prefixed flex class to the documentation

### CODE REVIEW

File: _lib/solid-base/_deprecated.scss [hunk 1]
```diff
@@ -1,7 +1,11 @@
...
-// @TODO Remove in 3.0
-// Deprecated Messaging
-// --------------------------------------------------
+// Deprecated SCSS Classes
+// Classes that would cause breaking changes if removed
+// ----------------------------------------------------------
+
+
+// Page Messaging:
+// @TODO Remove in 3.0 (arbitrary)
```
Comment (emilybrick) [inline _lib/solid-base/_deprecated.scss]: idk if these to-do's are necessary (or right), we chose random releases to essentially cut the class cold-turkey

Comment (JackWReid) [inline _lib/solid-base/_deprecated.scss]: seems real silly to me

Comment (JackWReid) [inline _lib/solid-base/_deprecated.scss]: i think we should ditch that stuff. it's so prone to getting ignored.

Comment (emilybrick) [inline _lib/solid-base/_deprecated.scss]: ok

Comment (emilybrick) [inline _lib/solid-base/_deprecated.scss]: we should have something in place in lieu of that, otherwise the deprecated stuff will be there indefinitely

Comment (JackWReid) [inline _lib/solid-base/_deprecated.scss]: Why don't we just commit to making a deprecation every time we major version?

**Comment by emilybrick** [comment]
@JackWReid updated

**Pull request merged**